### DISCUSSION TITLE
Can we have an installer log please

### DISCUSSION THREAD

**Discussion by mmtechconsulting** [MicrosoftEdge/WebView2Feedback#4029] [Q&A]
I have intermittent issues installing webview2 so i would appreciate an installer logging argument if possible to be able to troubleshoot.

Is this possible?

**Comment by ajtruckle** [comment]
I like this idea. Ideally we would be able to inject the log output into our own installer logs if possible. I use Inno Setup. +1 from me! 😊

**Comment by LiangTheDev** [comment]
Let me ask Edge installer team. Edge installer always generates logs, is it possible for you to read contents from that log? The location of installer log file is at https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/install.md#installer-logs

**Comment by mmtechconsulting** [comment]
> Let me ask Edge installer team. Edge installer always generates logs, is it possible for you to read contents from that log? The location of installer log file is at https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/install.md#installer-logs

So the webview2 installer writes the same logs as Edge?

**Comment by ajtruckle** [comment]
@mmtechconsulting I was about to ask the same question.

**Comment by LiangTheDev** [comment]
Yes, WebView2 installer logs to the same file as Edge.

The installer team stated that you can also specify a log file path by using --log-file argument. That's not what you're asking though :). There is no for logging into another log file owned by another app.

**Comment by ajtruckle** [comment]
@LiangTheDev But that is still useful. We can now provide our own WebView2 log file path for any post error checking. So it is a big help. Thanks. 👍 So, do you mean:

`path-to-setup.exe -log-file "path-to-log.txt"`

?

**Comment by mmtechconsulting** [comment]
I am asking all these because we got Webview2 installing during OSD, Operating system deployment. Webview2 intermittently fails during the task sequence so it is hard to identify where the installer fails without a log file. I have a script that picks up all the logs from the failed build and uploads them to a network share for review.
So Webview spews out an error that indicates that another installation might be going on or the product is already installed or something like that. I run a procmon during the task sequence and i can see that after the image is laid to the disk Edge talk back home and updates in the background. Then a few apps install and then Webview2 installs. There we got the intermittent failure. I have also noticed in procmon that Webview also talks home to get the latest updates.
Do you think that Edge trying to update might be causing the issue? I believe the log would give us more insight.

**Comment by mmtechconsulting** [comment]
> @LiangTheDev But that is still useful. We can now provide our own WebView2 log file path for any post error checking. So it is a big help. Thanks. 👍 So, do you mean:
> 
> `path-to-setup.exe -log-file "path-to-log.txt"`
> 
> ?

Is this working @ajtruckle  have you by any chance tested it?

**Comment by ajtruckle** [comment]
@mmtechconsulting Tried in a console window yesterday and could not work out the right syntax. The UI kept appearing with an error. 

There is no help built in detailing command line switches / usage.

**Comment by LiangTheDev** [comment]
Chromium command line switch normally work like the following, have you tried that?
`path-to-setup.exe --log-file=path-to-log.txt`

**Comment by ajtruckle** [comment]
@LiangTheDev For me, when I try that with the runtime exe I have it shows the GUI with an error.  I assume you tried too? 🤔

**Comment by LiangTheDev** [comment]
Tried it and got an error. Let me ask Edge installer team.

**Comment by LiangTheDev** [comment]
Edge installer team agreed that there is currently no way to specify the log file path. You would have to use the Edge installer log file. It seems that you are already doing that.
FWIW, the `--log-path` was a switch for the inter setup.exe that is doing the real installation of the WebView2 Runtime, not exposed in the final installer executable.
